WalletWasabi/Blockchain/Analysis/BlockchainAnalyzer.cs

```diff
@@ -99,36 +103,19 @@ private int Intersect(IEnumerable<int> anonsets, double coefficient)
 		return normalizedIntersectionAnonset;
 	}
 
-	private void AnalyzeCoinjoin(SmartTransaction tx, int newInputAnonset, ISet<HdPubKey> distinctWalletInputPubKeys)
+	private void AnalyzeCoinjoin(SmartTransaction tx, int newInputAnonset, int sanctionedInputAnonset)
 	{
-		var indistinguishableWalletOutputs = tx
-			.WalletOutputs.GroupBy(x => x.Amount)
-			.ToDictionary(x => x.Key, y => y.Count());
-
-		var indistinguishableOutputs = tx.Transaction.Outputs
-			.GroupBy(x => x.ScriptPubKey)
-			.ToDictionary(x => x.Key, y => y.Sum(z => z.Value))
-			.GroupBy(x => x.Value)
-			.ToDictionary(x => x.Key, y => y.Count());
-
-		var inputCount = tx.Transaction.Inputs.Count;
-		var ownInputCount = tx.WalletInputs.Count;
+		var foreignInputCount = tx.ForeignInputs.Count;
 
 		foreach (var newCoin in tx.WalletOutputs)
 		{
-			var output = newCoin.TxOut;
-			var equalOutputCount = indistinguishableOutputs[output.Value];
-			var ownEqualOutputCount = indistinguishableWalletOutputs[output.Value];
-
 			// Anonset gain cannot be larger than others' input count.
-			var anonset = Math.Min(equalOutputCount - ownEqualOutputCount, inputCount - ownInputCount);
-
 			// Picking randomly an output would make our anonset: total/ours.
```

Comment:
Let's remove the comment about picking, since that's done inside `ComputeAnonymityContribution`.